### ISSUE TITLE
Investigate why CI or check-cxx with ubsan does not detect the issue

### ISSUE THREAD

**Issue by vitalybuka** [llvm/llvm-project#152184]
> #152181 should fix the issue. I'd really like to know why this wasn't caught in the CI though. Is there some special flag required to check for this kind of stuff?

I believe CI has check-cxx with ubsan, our bot has "check-cxx with ubsan" and it does not fail.

What fails is using libcxx to compile llvm. Type from report is:
```
std::map<std::string, std::unique_ptr<ACLEIntrinsic>> ACLEIntrinsics;
```

So I guess libcxx lack of test coverage for them?

_Originally posted by @vitalybuka in https://github.com/llvm/llvm-project/issues/152180#issuecomment-3156013122_

**Comment by vitalybuka** [comment]
From discord by @philnik777:

 I've figured out why we don't get any errors in our CI: clang++: warning: the object size sanitizer has no effect at -O0, but is explicitly enabled: -fsanitize=object-size [-Winvalid-command-line-argument]. I wonder whether we should just enable UBSan in our optimized build...
